RecoPixelVertexing/PixelTriplets/plugins/BrokenLineFitOnGPU.cc

```diff
@@ -3,67 +3,118 @@
 void HelixFitOnGPU::launchBrokenLineKernelsOnCPU(HitsView const* hv, uint32_t hitsInFit, uint32_t maxNumberOfTuples) {
   assert(tuples_);
 
+#ifdef BROKENLINE_DEBUG
+  setlinebuf(stdout);
+#endif
+
   //  Fit internals
-  auto hitsGPU_ =
-      std::make_unique<double[]>(maxNumberOfConcurrentFits_ * sizeof(riemannFit::Matrix3xNd<4>) / sizeof(double));
-  auto hits_geGPU_ =
-      std::make_unique<float[]>(maxNumberOfConcurrentFits_ * sizeof(riemannFit::Matrix6x4f) / sizeof(float));
-  auto fast_fit_resultsGPU_ =
+  auto tkidGPU = std::make_unique<caConstants::tindex_type[]>(maxNumberOfConcurrentFits_);
+  auto hitsGPU =
+      std::make_unique<double[]>(maxNumberOfConcurrentFits_ * sizeof(riemannFit::Matrix3xNd<6>) / sizeof(double));
+  auto hits_geGPU =
+      std::make_unique<float[]>(maxNumberOfConcurrentFits_ * sizeof(riemannFit::Matrix6xNf<6>) / sizeof(float));
+  auto fast_fit_resultsGPU =
       std::make_unique<double[]>(maxNumberOfConcurrentFits_ * sizeof(riemannFit::Vector4d) / sizeof(double));
 
   for (uint32_t offset = 0; offset < maxNumberOfTuples; offset += maxNumberOfConcurrentFits_) {
     // fit triplets
-    kernel_BLFastFit<3>(
-        tuples_, tupleMultiplicity_, hv, hitsGPU_.get(), hits_geGPU_.get(), fast_fit_resultsGPU_.get(), 3, offset);
+    kernel_BLFastFit<3>(tuples_,
+                        tupleMultiplicity_,
+                        hv,
+                        tkidGPU.get(),
+                        hitsGPU.get(),
+                        hits_geGPU.get(),
+                        fast_fit_resultsGPU.get(),
+                        3,
+                        3,
+                        offset);
 
     kernel_BLFit<3>(tupleMultiplicity_,
                     bField_,
                     outputSoa_,
-                    hitsGPU_.get(),
-                    hits_geGPU_.get(),
-                    fast_fit_resultsGPU_.get(),
-                    3,
-                    offset);
-
-    // fit quads
-    kernel_BLFastFit<4>(
-        tuples_, tupleMultiplicity_, hv, hitsGPU_.get(), hits_geGPU_.get(), fast_fit_resultsGPU_.get(), 4, offset);
-
-    kernel_BLFit<4>(tupleMultiplicity_,
-                    bField_,
-                    outputSoa_,
-                    hitsGPU_.get(),
-                    hits_geGPU_.get(),
-                    fast_fit_resultsGPU_.get(),
-                    4,
-                    offset);
+                    tkidGPU.get(),
+                    hitsGPU.get(),
+                    hits_geGPU.get(),
+                    fast_fit_resultsGPU.get());
 
     if (fit5as4_) {
```

Comment:
I guess this should be renamed (also later) to something like `fit_all_as_4_` ?

Reply:
Yeah, I call it fitNas4.
I did not want to change configurable at this moment (not sure what is the state of HLT conf)